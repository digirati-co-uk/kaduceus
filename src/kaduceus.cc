#include "kaduceus.h"

namespace digirati::kaduceus {

std::unique_ptr<CxxKakaduImage> create_kakadu_image_reader(std::shared_ptr<CxxKakaduContext> ctx, rust::Box<AsyncReader> reader)
{
    return std::make_unique<CxxKakaduImage>(ctx, reader);
}

std::shared_ptr<CxxKakaduContext> create_kakadu_context()
{
    return std::make_shared<CxxKakaduContext>();
}

}
