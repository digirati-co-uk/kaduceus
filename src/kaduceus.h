#ifndef KAKADU_RS_H
#define KAKADU_RS_H

#include "rust/cxx.h"
#include "data_source.h"
#include "kakadu_context.h"
#include "kakadu_decompressor.h"
#include "kakadu_image_reader.h"

#endif // KAKADU_RS_H
