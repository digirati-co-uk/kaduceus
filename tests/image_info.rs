use std::path::{Path, PathBuf};

use insta::assert_debug_snapshot;
use kaduceus::{Info, KakaduContext, KakaduImageReader};

fn image_path(name: &str) -> PathBuf {
    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
    path.push(name);

    path
}

fn image_info<P: AsRef<Path>>(path: P) -> Info {
    let path = path.as_ref();
    let rt = tokio::runtime::Builder::new_current_thread()
        .build()
        .expect("test must not be run from an async context");

    let filename = path
        .file_name()
        .map(|str| str.to_string_lossy())
        .unwrap_or("unknown".into());

    let file = rt.block_on(async move { tokio::fs::File::open(path).await.unwrap() });

    let ctx = KakaduContext::default();
    let mut reader = KakaduImageReader::new(ctx, file, Some(filename.into()));

    reader.info()
}

fn test_image_info(name: &str) {
    let mut settings = insta::Settings::clone_current();
    settings.set_snapshot_suffix(name);

    let _guard = settings.bind_to_scope();

    let image_path = image_path(name);
    let image_info = image_info(image_path);

    assert_debug_snapshot!(image_info);
}

#[test]
fn test_reference_image() {
    test_image_info("testdata/reference.jp2");
}
