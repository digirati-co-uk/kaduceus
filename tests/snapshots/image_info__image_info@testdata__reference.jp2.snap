---
source: tests/image_info.rs
expression: image_info
---
Info {
    width: 2717,
    height: 3701,
    tile_width: 1024,
    tile_height: 1024,
    dwt_levels: 5,
}
